docs/hello_nextflow/03_hello_containers.md

```diff
@@ -1,82 +1,442 @@
+/ # ls data
+bam              greetings.csv    ref              sample_bams.txt  samplesheet.csv
+```
+
+### 2.2. Use the mounted data
+
+Now that we have mounted the `data` directory into the container, we can use the `cowsay` command to display the contents of the `greetings.csv` file.
+
+```bash
+cat data/greetings.csv | cowsay
+```
+
+Output:
+
+```console title="Output"
+ _____________________
+< Hello,Bonjour,Holà >
+ ---------------------
+        \   ^__^
+         \  (oo)\_______
+            (__)\       )\/\
+                ||----w |
+                ||     ||
+/ # exit
+```
+
+### Takeaway
+
+You know how to pull a container and run it interactively, and you know how to use that to try out commands without having to install any software on your system.
+
+### What's next?
+
+Learn how to get a container image for any pip/conda-installable tool.
 
 ---
 
-## 2. Use the container in a workflow
+## 2. Get a container image for a pip/conda-installable tool
+
+Some software developers provide container images for their software that are available on container registries like Docker Hub, but many do not.
+New Docker images can be created by writing a `Dockerfile` that specifies how to build the image, but this can be a lot of work.
+Instead, we can use the Seqera Containers web service to create a container image for us.
+
+### 2.1. Navigate to Seqera Containers
+
+Navigate to [Seqera Containers web service](https://www.seqera.io/containers/) and search for the `quote` pip package.
+
+![Seqera Containers](img/seqera-containers-1.png)
+
+### 2.2. Request a container image
+
+Click on "+Add" and then "Get Container" to request a container image for the `quote` pip package.
+
+![Seqera Containers](img/seqera-containers-2.png)
+
+If this is the first time a community container has been built for this package, it may take a few minutes to complete.
+Click to copy the URI (e.g. `community.wave.seqera.io/library/pip_quote:ae07804021465ee9`) of the container image that was created for you.
+
+### 2.3. Use the container image
 
-TODO
+You can now use the container image to run the `quote` command and get a random saying from Grace Hopper.
 
-### 2.1. [step 1]
+```bash
+docker run --rm community.wave.seqera.io/library/pip_quote:ae07804021465ee9 quote "Grace Hopper"
+```
+
+Output:
+
+```console title="Output"
+Humans are allergic to change. They love to say, 'We've always done it
+this way.' I try to fight that. That's why I have a clock on my wall
+that runs counter-clockwise.
+```
 
-TODO
+### 2.4. STRETCH GOAL: Build the container image yourself
 
-### 2.2. [step 2]
+Go back to the Seqera Containers website and click on the "Build Details" button.
+There you can see the details of how our container image was built.
+You can view the conda environment file and the `Dockerfile` together make up the recipe for the container image.
```

Comment:
sentence doesn't quite work? Did you mean "You can see that"?

Reply:
Fixed.